### ISSUE TITLE
Re-enable the BlockVerifier continuous_blockchain_test

### ISSUE THREAD

**Issue by teor2345** [ZcashFoundation/zebra#1011]
To handle out of order blocks, we decided to implement a temporary out-of-order block cache in the state service, and disable some of the affected tests.

We need to re-enable these tests, once we have implemented the features they depend on.

## TODO
- [ ] Re-enable `zebra_consensus::chain::tests::continuous_blockchain_test`, which depends on contextual verification of post-Sapling blocks (after #1135)

### Context

* we ignored some unit tests in PR #1012 
* we ignored some unit tests in commit c042555ac714819b03efbc565bc0223fedf43102 - `Temporarily ignore failing state tests`

**Comment by teor2345** [comment]
@hdevalence, @yaahc I think the changes in https://github.com/ZcashFoundation/zebra/pull/902#issuecomment-688616536 mainly affect this ticket, rather than PR #1012 

#1012 changes the verifiers so they stop trying to handle out of order blocks themselves. This ticket makes the state handle those out of order blocks.

**Comment by hdevalence** [comment]
I don't think we want a single queue, because the two queues are used in different places and contain disjoint data.  Two queues would be a better fit.

**Comment by teor2345** [comment]
#1101 does not resolve this issue - those tests aren't enabled yet.

**Comment by teor2345** [comment]
Moved out-of-order transcripts to the state tracking issue #1049.
Moved sync tests to #1188.

The remaining work is out of scope for the first alpha.

**Comment by mpguerra** [comment]
Seems like this should have been closed by #1101 but wasn't?

Closing manually, feel free to re-open if this is not the case

**Comment by mpguerra** [comment]
Oops, just checked history, re-opening, sorry for the noise

**Comment by teor2345** [comment]
This task can't be done in the first alpha, because it depends on the refactor in #1135, which is long-term work.

**Comment by teor2345** [comment]
It seems like this test got removed or renamed a long while ago.
